src/dtls_connector.rs

```diff
@@ -125,22 +146,37 @@ impl DtlsConnector {
         &self,
         domain: &str,
         stream: S,
-    ) -> Result<DtlsStream<S>, HandshakeError<S>>
+    ) -> Result<SyncDtlsStream<S>, HandshakeError<S>>
     where
         S: io::Read + io::Write,
     {
-        let mut ssl = self
-            .connector
-            .configure()?
-            .use_server_name_indication(self.use_sni)
-            .verify_hostname(!self.accept_invalid_hostnames);
-        if self.accept_invalid_certs {
-            ssl.set_verify(SslVerifyMode::NONE);
-        }
+        let conf = self.configure()?;
+        let stream = conf.connect(domain, stream)?;
 
-        let stream = ssl.connect(domain, stream)?;
-        Ok(DtlsStream::from(stream))
+        Ok(SyncDtlsStream::from(stream))
     }
+
+    #[cfg(feature="async")]
+    pub async fn async_connect<S: fmt::Debug>(
+        &self,
+        domain: &str,
+        stream: S,
+    ) -> Result<AsyncDtlsStream<S>, AsyncConnectError<S>>
+    where
+        S: AsyncRead + AsyncWrite + Unpin
+    {
+        let conf = self.configure().map_err(AsyncConnectError::ErrorStack)?;
+        let stream = tokio_openssl::connect(conf, domain, stream).await.map_err(AsyncConnectError::TokioOpenSsl)?;
+
+        Ok(AsyncDtlsStream::from(stream))
+    }
+}
+
+#[cfg(feature="async")]
+#[derive(Debug)]
+pub enum AsyncConnectError<S> {
```

Comment:
Move this over to error.rs

Reply:
I left it here, for a moment by purpose. The way it is now is not too elegant, and I wanted to ask you for an opinion. Actually, I can see three alternatives here:
1) move AsyncConnectError to error.rs and let it be like this
2) extend error::Error with another choice tokio_openssl::HandshakeError
3) change both connect / async_connect error option to something like Box<dyn std::result::Error> so it will be able to accommodate anything.

Reply:
I moved AsyncConnectError to error.rs, although alternatives mentioned above are still valid